src/colab/client.ts

```diff
@@ -275,6 +278,63 @@ export class ColabClient {
     );
   }
 
+  /**
+   * Propagates credentials to the backend.
+   *
+   * @param endpoint - The assignment endpoint to propagate credentials to.
+   * @param authType - If `dfs_persistent`, backend will record the credentials
+   *   and avoid future requests.
+   * @param fileId - This notebook's file ID.
+   * @param renderDataToken - A URL-safe, tamper-proof token containing a record
+   *   of FE rendering data, in the form of an ARI `AuditRenderData` proto
+   *   message, that can be used to reconstruct the text that was shown to users
+   *   during consent moments.
+   * @param dryRun - If true, check if credentials are already propagated to the
+   *   backend and/or obtain an OAuth redirect URL.
+   * @param signal - Optional {@link AbortSignal} to cancel the request.
+   * @returns Whether propagation is successful. If not, an OAuth redirect URL
+   *   is returned to obtain the credentials.
+   */
+  async propagateCredentials(
```

Comment:
Nit: include _drive_ in method name.

```suggestion
  async propagateDriveCredentials(
```